starter_code/elevator.js

```diff
@@ -3,17 +3,68 @@ class Elevator {
     this.floor      = 0;
     this.MAXFLOOR   = 10;
     this.requests   = [];
+    this.direction = "";
+    this.timerId;
+    this.waitingList = []; //made request but not on yet
+    this.passengers = []; //currently in elevator
   }
 
-  start() { }
-  stop() { }
-  update() { }
-  _passengersEnter() { }
-  _passengersLeave() { }
-  floorUp() { }
-  floorDown() { }
-  call() { }
-  log() { }
+  start() {
+    this.timerId = setInterval(update, 1000);
+  }
+
+  stop() {
+    this.timerId = clearInterval(update, 0);
+  }
+
+  update() {
+    log();
+  }
+
+  _passengersEnter() {
+    for (var i = 0; i < waitingList.length; i++) {
```

Comment:
it would be better to use a forEach here with an arrow function
    **this.waitingList.forEach((person, index) => {**